### PR TITLE
Fixed wrong url redirect when edit product review from product view page

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

wrong url redirect when edit product review from product view page
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. Catalog-> product-> Product Reviews 
- edit any review -> click on save

### Steps to reproduce

1. Catalog-> product-> Product Reviews 
- edit any review -> click on save

### Expected result

1. After save redirect product page 

### Actual result

1. After save redirect review -> product -> index

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by rogyar** [comment]
Hi @ravi-chandra3197. Thank you for your contribution. The system is designed in that way: after review/customer some other entity is saved you are being redirected to the entities grid. So I'm not sure about an improvement we implement in the current PR since there's nothing wrong with the current URL and redirect.

**Comment by ravi-chandra3197** [comment]
> Hi @ravi-chandra3197. Thank you for your contribution. The system is designed in that way: after review/customer some other entity is saved you are being redirected to the entities grid. So I'm not sure about an improvement we implement in the current PR since there's nothing wrong with the current URL and redirect.

Hello @rogyar 
when we edit product review from the product view page where we have 1000 review, so in the current scenario, after review edit, it will redirect to review grid, now we want to edit more review from product or edit product entity so it's very hard to manage or not a user-friendly workflow.

**Comment by rogyar** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rogyar. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rogyar, here is your [new Magento instance](https://pr-22202.instances.magento-community.engineering).
Admin access: [https://pr-22202.instances.magento-community.engineering/admin](https://pr-22202.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by rogyar** [comment]
@ravi-chandra3197 thank you for the clarification. Now I get your point.

File: app/code/Magento/Review/Controller/Adminhtml/Product/Save.php [hunk 1]
```diff
@@ -73,6 +73,10 @@ public function execute()
             } else {
                 $resultRedirect->setPath('*/*/');
             }
+            $productId = $this->getRequest()->getParam('productId');
```
Comment (rogyar) [inline app/code/Magento/Review/Controller/Adminhtml/Product/Save.php]: Please use type casting to `(int)` here to limit cases of passing unpredictable values to the URL.

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Review/Controller/Adminhtml/Product/Save.php (hunk 1 above)
Comment (ravi-chandra3197) [reply app/code/Magento/Review/Controller/Adminhtml/Product/Save.php]: Hello @rogyar 
I have updated code as per suggestion

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-4719](https://jira.corp.magento.com/browse/ENGCOM-4719) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**